test/celeritas/field/FieldPropagator.test.cc

```diff
@@ -1176,6 +1182,83 @@ TEST_F(SimpleCmsTest, electron_stuck)
     }
 }
 
+TEST_F(SimpleCmsTest, vecgeom_failure)
+{
+    UniformZField      field(1000);
+    FieldDriverOptions driver_options;
+
+    auto geo = this->init_geo({1.23254142755319734e+02,
+                               -2.08186543568394598e+01,
+                               -4.08262349901495583e+01},
+                              {-2.59700373666105766e-01,
+                               -8.11661685885768147e-01,
+                               -5.23221772848529443e-01});
+
+    auto calc_radius
+        = [&geo]() { return std::hypot(geo.pos()[0], geo.pos()[1]); };
+
+    bool successful_reentry = false;
+    {
+        auto particle
+            = this->init_particle(this->particle()->find(pdg::electron()),
+                                  MevEnergy{3.27089632881079409e-02});
+        auto stepper = make_mag_field_stepper<DiagnosticDPStepper>(
+            field, particle.charge());
+        auto propagate
+            = make_field_propagator(stepper, driver_options, particle, &geo);
+        auto result = propagate(1.39170198361108938e-05);
+        EXPECT_EQ(result.boundary, geo.is_on_boundary());
+        EXPECT_EQ("em_calorimeter", this->volume_name(geo));
+        EXPECT_SOFT_EQ(125.00000000000001, calc_radius());
+        EXPECT_EQ(2, stepper.count());
+    }
+    {
+        ASSERT_TRUE(geo.is_on_boundary());
+        // Simulate MSC making us reentrant
+        geo.set_dir({-1.31178657592616127e-01,
+                     -8.29310561920304168e-01,
+                     -5.43172303859124073e-01});
+        geo.cross_boundary();
+        successful_reentry = (this->volume_name(geo) == "em_calorimeter");
+        if (!CELERITAS_USE_VECGEOM
+            || !starts_with(celeritas_vecgeom_version, "1.1."))
+        {
+            // Both ORANGE and newer VecGeom versions should successfully
+            // reenter. Some older versions on some systems will sometimes move
+            // into the world volume.
+            EXPECT_EQ("em_calorimeter", this->volume_name(geo));
```

Comment:
In my test with VecGeom v1.2.0, this line is failing: Failure
Expected equality of these values:
  "em_calorimeter"
  this->volume_name(geo)
    Which is: "world"

Reply:
Oh dear, that means it's something *besides* the vecgeom version causing this change. On the CI with 1.1.18 it returns "world", but not on my machine. I had assumed it was a change upstream that caused the boundary behavior to change. 🤔